### ISSUE TITLE
[smile_module_record] (12.0) Invalid field 'res_field' in leaf

### ISSUE THREAD

**Issue by intelliping** [Smile-SA/odoo_addons#44]
```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/ERP/odoo/http.py", line 654, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/ERP/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/ERP/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/ERP/odoo/http.py", line 696, in dispatch
    result = self._call_function(**self.params)
  File "/ERP/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/ERP/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/ERP/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/ERP/odoo/http.py", line 939, in __call__
    return self.method(*args, **kw)
  File "/ERP/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/ERP/addons/web/controllers/main.py", line 966, in call_button
    action = self._call_kw(model, method, args, {})
  File "/ERP/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/ERP/odoo/api.py", line 749, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/ERP/odoo/api.py", line 736, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/ERP/my-modules/smile_module_record/wizard/base_module_export.py", line 276, in button_create_module
    self.create_module()
  File "<decorator-gen-99>", line 2, in create_module
  File "/ERP/odoo/api.py", line 372, in loop
    result = [method(rec, *args, **kwargs) for rec in self]
  File "/ERP/odoo/api.py", line 372, in <listcomp>
    result = [method(rec, *args, **kwargs) for rec in self]
  File "/ERP/my-modules/smile_module_record/wizard/base_module_export.py", line 230, in create_module
    datas = self._get_data_filecontent()
  File "/ERP/my-modules/smile_module_record/wizard/base_module_export.py", line 183, in _get_data_filecontent
    for model, rows in self._export_data_by_model():
  File "/ERP/my-modules/smile_module_record/wizard/base_module_export.py", line 109, in _export_data_by_model
    recs = res_obj.search(res_obj._log_access and domain or [])
  File "/ERP/odoo/models.py", line 1561, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/ERP/odoo/models.py", line 4078, in _search
    query = self._where_calc(args)
  File "/ERP/odoo/models.py", line 3869, in _where_calc
    e = expression.expression(domain, self)
  File "/ERP/odoo/osv/expression.py", line 672, in __init__
    self.parse()
  File "/ERP/odoo/osv/expression.py", line 853, in parse
    raise ValueError("Invalid field %r in leaf %r" % (left, str(leaf)))
ValueError: Invalid field 'res_field' in leaf "<osv.ExtendedLeaf: ('res_field', '=', False) on ir_config_parameter (ctx: )>"
```

**Comment by intelliping** [comment]
Turns out the issue is due to missing dependency of **calendar**. Suggest to add **calendar** as dependency in the manifest.

**Comment by ralienpp** [comment]
I confirm that that is the case, this happened in Odoo v.12 and adding the `calendar` app solved it.

**Comment by isabellerichard** [comment]
Hi, thanks for your analysis. We will fix it as soon as we test your solution.

**Comment by isabellerichard** [comment]
Hi @LipingSun @ralienpp ,

Bug was fixed in commit fa5632be469d6895955d8bba66468df0d51aabef.

Could you test please?
